DragaliaAPI/DragaliaAPI/Features/StorySkip/StorySkipService.cs

```diff
@@ -0,0 +1,191 @@
+using DragaliaAPI.Database.Entities;
+using DragaliaAPI.Database.Repositories;
+using DragaliaAPI.Features.Dungeon;
+using DragaliaAPI.Features.Fort;
+using DragaliaAPI.Features.Quest;
+using DragaliaAPI.Features.Reward;
+using DragaliaAPI.Models.Generated;
+using DragaliaAPI.Services;
+using DragaliaAPI.Services.Game;
+using DragaliaAPI.Shared.Definitions.Enums;
+using DragaliaAPI.Shared.Features.StorySkip;
+using DragaliaAPI.Shared.MasterAsset;
+using DragaliaAPI.Shared.MasterAsset.Models.QuestRewards;
+using static DragaliaAPI.Shared.Features.StorySkip.FortConfigurations;
+
+namespace DragaliaAPI.Features.StorySkip;
+
+public class StorySkipService(
+    IFortRepository fortRepository,
+    IFortService fortService,
+    ILogger<StorySkipService> logger,
+    IQuestCompletionService questCompletionService,
+    IQuestRepository questRepository,
+    IQuestTreasureService questTreasureService,
+    IRewardService rewardService,
+    IStoryService storyService,
+    IUpdateDataService updateDataService,
+    IUserDataRepository userDataRepository
+) : IStorySkipService
+{
+    private async Task<QuestMissionStatus> CompleteQuestMissions(int questId, bool[] currentState)
+    {
+        List<AtgenMissionsClearSet> clearSet = new();
+
+        bool[] newState = { true, true, true };
+
+        QuestRewardData rewardData = MasterAsset.QuestRewardData[questId];
+        for (int i = 0; i < 3; i++)
+        {
+            (QuestCompleteType type, int value) = rewardData.Missions[i];
+
+            if (!currentState[i])
+            {
+                newState[i] = true;
```

Comment:
I don't think this does anything

Reply:
Yes, I think you're right. I copy/pasted this from the mission completion service. I made just the essential changes, but I can clean it up.